Confuser.Runtime/IntegrityProtection.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Confuser.Runtime
+{
+    internal static class IntegrityProtection
+    {
+        internal static void Initialize()
+        {
+            var bas = new StreamReader(typeof(IntegrityProtection).Assembly.Location).BaseStream;
+            var file = new BinaryReader(bas);
+            var file2 = File.ReadAllBytes(typeof(IntegrityProtection).Assembly.Location);
+
+            var byt = file.ReadBytes(file2.Length - 32);
+            var a = Hash(byt);
+            file.BaseStream.Position = file.BaseStream.Length - 32;
+            string b = Encoding.ASCII.GetString(file.ReadBytes(32));
+
+            if (a != b)

```

Comment:
I still haven't pushed this PR because there is a huge security flaw here : what prevents me from decompiling the protected file and NOP this if statement ? @rexyrexy